app/connectors/MessageConnector.scala

```diff
@@ -47,4 +47,17 @@ class MessageConnector @Inject()(http: HttpClient, appConfig: AppConfig) extends
     val url = rootUrl + s"/arrivals/${Utils.urlEncode(arrivalId)}/messages"
     http.POSTString(url, message, requestHeaders())(CustomHttpReader, implicitly, implicitly)
   }
+
+  def getArrivalMessages(arrivalId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[HttpResponse, Arrival]] = {
+    val url = rootUrl + s"/arrivals/${Utils.urlEncode(arrivalId)}/messages"
+    http.GET[HttpResponse](url, queryParams = Seq(), responseHeaders())(CustomHttpReader, implicitly, implicitly).map { r =>
+      if(is2xx(r.status)) {
+        r.json.asOpt[Arrival] match {
+          case Some(x) => Right(x)
+          case _ => Left(CustomHttpReader.recode(INTERNAL_SERVER_ERROR, r))
+        }
+      } else Left(r)
+    }
+  }
```

Comment:
I think the short named variables in the PR should be changed to the full name representing the type.

Reply:
I'm following the pattern that I followed previously in this file. I'll change those as well.

Reply:
Actually, I think I have something I can deduplicate here, so I might try that.